### PR TITLE
60 maturation of acs ecs

### PR DESCRIPTION
**Pull request by dccutrig**

### CODE REVIEW

File: src/src_main/crypto.c [hunk 1]
```diff
@@ -21,6 +21,8 @@
 */
 #include "crypto.h"
 
+
+#include <gcrypt.h> // GCRY acs/ecs nums
```
Comment (IbraheemYSaleh) [inline src/src_main/crypto.c:25]: The libgcrypt interface should be the only thing that depends on libgcrypt -- the libgcrypt dependency should not be a core CryptoLib dependency. If you import libgcrypt here, then CryptoLib must have libgcrypt in order to be built and violates that idea. We should have our own enums that map to the libgcrypt enums when using the libgcrypt interface, or the KMC Crypto algorithms when using that  interface.

File: include/crypto.h [hunk 2]
```diff
@@ -111,6 +111,7 @@ uint16_t Crypto_Calc_FECF(uint8_t* ingest, int len_ingest);
 void Crypto_Calc_CRC_Init_Table(void);
 uint16_t Crypto_Calc_CRC16(uint8_t* data, int size);
 int32_t Crypto_Check_Anti_Replay(SecurityAssociation_t *sa_ptr, uint8_t *arsn, uint8_t *iv);
+int32_t Crypto_Get_Acs_Algo(int8_t algo_enum);
```
Comment (IbraheemYSaleh) [inline include/crypto.h:114]: This function either needs to not depend on libgcrypt enums, or be moved into the cryptography interface where it has access to them.

**Review by IbraheemYSaleh — CHANGES_REQUESTED** [review]
This branch makes libgcrypt a core dependency -- that should not be the case and needs to properly isolate libgcrypt-isms to the libgcrypt cryptography interface implementation.

File: include/crypto.h (hunk 2 above)
Comment (dccutrig) [reply include/crypto.h:114]: All noted, thanks for the feedback!

**Review by IbraheemYSaleh — APPROVED** [review]

**Pull request merged**